R/plot_breakpoints.R

```diff
@@ -1,24 +1,72 @@
 #' Function to plot breakpoints of a gene in a sample/cohort
 
 #' @param domainDataFrame A dataframe from star fusion or arriba standardized to run through the filtering steps and get_Pfam_domain() function to get pfam domain annotation per gene
-#' @param exons exon level information from gtf
-#' @param geneposition Left/Right position of gene
+#' @param exons exon level information from gtf availbale in the package exonsToPlot.RDS
+#' @param geneposition Position of gene in the fusion allowed values are Left (Gene1A) or Right (Gene1B)
+#' @param sampleid id (if multiple provide a character vector) from Sample column of standardized fusion file format
+#' @param fusionname value from FusionName column of standardized fusion file format (Gene1A--Gene1B)
+#' @param leftBreakpoint If a specific breakpoint needs to be plotted then provide genomic coordinate as LeftBreakpoint
+#' @param rightBreakpoint  If a specific breakpoint needs to be plotted then provide genomic coordinate as RightBreakpoint
+#' @param base_size base size for text in plot ; default is 12
 #'
 #' @export
 #'
 #' @return ggplot of gene and breakpoints
 #'
 #' @examples
-#' # TODOTODO
-plot_breakpoints <- function(domainDataFrame = domainDataFrame,
+#' standardFusionfile <- system.file("extdata", "PutativeOncogenicFusion.tsv", package = "annoFuse")
+#' standardFusioncalls <- readr::read_tsv(standardFusionfile)
+#' exons <- readRDS(system.file("extdata", "exonsToPlot.RDS", package = "annoFuse"))
+#' bioMartDataPfam <- readRDS(system.file("extdata", "pfamDataBioMart.RDS", package = "annoFuse"))
+#' domainDataFrame <- get_Pfam_domain(standardFusioncalls = standardFusioncalls, bioMartDataPfam = bioMartDataPfam, keepPartialAnno = TRUE)
+#' left <- plot_breakpoints(sampleid = "BS_044XZ8ST", domainDataFrame = domainDataFrame, exons = exons, geneposition = "Left", fusionname = "ANTXR1--BRAF")
+#' right <- plot_breakpoints(sampleid = "BS_044XZ8ST", domainDataFrame = domainDataFrame, exons = exons, geneposition = "Right", fusionname = "ANTXR1--BRAF")
+#' ggpubr::ggarrange(left, right, align = "h")
+plot_breakpoints <- function(domainDataFrame = NULL,
                              exons = exons,
-                             geneposition = geneposition) {
+                             geneposition = NULL,
+                             sampleid = NULL,
+                             fusionname = NULL,
+                             leftBreakpoint = NULL,
+                             rightBreakpoint = NULL,
```

Comment:
I can work on the default values once it is merged with the `exemplary` branch
I'm gonna need your expertise there to clarify what some meaningful defaults would be (or if not default values, the ones we would suggest when we write the docs ;) )